internal/persistence/ent/client/package.go

```diff
@@ -130,3 +132,56 @@ func (c *client) UpdatePackage(
 
 	return convertPkgToPackage(p), nil
 }
+
+// ListPackages returns a slice of packages filtered according to f.
+func (c *client) ListPackages(ctx context.Context, f persistence.PackageFilter) (
+	[]*datatypes.Package, *persistence.Page, error,
+) {
+	res := []*datatypes.Package{}
+
+	page, whole := filterPackages(c.ent.Pkg.Query(), &f)
+
+	r, err := page.All(ctx)
+	if err != nil {
+		return nil, nil, newDBError(err)
+	}
+
+	for _, i := range r {
+		res = append(res, convertPkgToPackage(i))
+	}
+
+	total, err := whole.Count(ctx)
+	if err != nil {
+		return nil, nil, newDBError(err)
+	}
+
+	pp := &persistence.Page{
+		Limit:  f.Limit,
+		Offset: f.Offset,
+		Total:  total,
+	}
+
+	return res, pp, err
+}
+
+// filterPackages filters a package query based on filtering inputs.
+func filterPackages(q *db.PkgQuery, f *persistence.PackageFilter) (page, whole *db.PkgQuery) {
+	h := NewFilter(q, SortableFields{
```

Comment:
Why `h`?

Reply:
I'm not sure — I copied it over from the STE implementation as is.

Reply:
I'm open to other suggestions.

Reply:
Maybe changing the parameter var name too? 

```
func filterPackages(q *db.PkgQuery, pf *persistence.PackageFilter) (page, whole *db.PkgQuery) {
	f := NewFilter(q, SortableFields{...
```